src/Persistent/Api.hs

```diff
@@ -198,16 +242,16 @@ type AddDb m a = UserWithProto a -> PersistM m => m a
 -- It could make sense for the traversal to point to more than one target for instance
 -- to index the record at different locations. For instance we could keep an additional
 -- global map of the comments, votes, likes and still call @addDb@ only once.
-addDb :: (HasMetaInfo a, FromProto a) => AulaTraversal (AMap a) -> AddDb m a
+addDb :: (HasMetaInfo a, FromProto a) => (AUID a -> DbField a) -> AddDb m a
 addDb l (cUser, pa) = do
+    a <- fromProto pa <$> nextMetaInfo cUser
     assertPersistM $ do
         db <- getDb id
-        let len = lengthOf l db
+        let len = undefined + 17 -- TODO: lengthOf (dbFieldSetter (l $ a ^. _Id) *and then go one level up*) db
```

Comment:
Here you got the limitation that `dbFieldSetter` is returning well ... a setter. At first we can disable this assertion.